libs/execution_context/ebpf_core.c

```diff
@@ -2073,6 +2073,25 @@ _ebpf_core_protocol_ring_buffer_map_async_query(
     return result;
 }
 
+static ebpf_result_t
+_ebpf_core_protocol_ring_buffer_map_write_data(_In_ const ebpf_operation_ring_buffer_map_write_data_request_t* request)
+{
+    ebpf_map_t* map = NULL;
+    ebpf_result_t result =
+        EBPF_OBJECT_REFERENCE_BY_HANDLE(request->map_handle, EBPF_OBJECT_MAP, (ebpf_core_object_t**)&map);
+    if (result != EBPF_SUCCESS) {
+        goto Exit;
+    }
+    if (ebpf_map_get_definition(map)->type != BPF_MAP_TYPE_RINGBUF) {
```

Comment:
I don't think this is sufficient since it would work with any ring buffer map, which I don't think is appropriate.
It should also check (say) a flag on the ring buffer map, or use a separate map type.

Reply:
Resolving per meeting discussion.  Will file separate issue.